src/components/DpInput/DpInput.vue

```diff
@@ -41,185 +42,81 @@
-    required: false,
-    default: ''
-  },
-
-  pattern: {
-    type: String,
-    required: false,
-    default: ''
-  },
-
-  placeholder: {
-    type: String,
-    required: false,
-    default: ''
-  },
-
-  /**
-   * Set to false to prevent default behavior onEnter.
-   */
-  preventDefaultOnEnter: {
-    type: Boolean,
-    required: false,
-    default: true
-  },
-
-  readonly: {
-    type: Boolean,
-    required: false,
-    default: null
-  },
-
-  required: {
-    type: Boolean,
-    required: false,
-    default: null
-  },
-
-  /**
-   * When setting a number for the `size` prop, this is directly rendered
-   * as html attribute on the input element. Also, it is assumed that visual sizing
-   * based on that value shall be applied, that is why both container classes
-   * and element classes do not define any width styles when a size is set here.
-   */
-  size: {
-    type: [Number, null],
-    required: false,
-    default: null
-  },
-
-  type: {
-    type: String,
-    required: false,
-    default: 'text'
-  },
-
-  value: {
-    type: String,
-    required: false,
-    default: ''
-  },
+// Define an interface for the props
+interface Props {
+  ariaLabelledby?: string | null
+  autocomplete?: string
+  dataCounter?: string
+  dataCy?: string
+  dataDpValidateError?: string
+  dataDpValidateErrorFieldname?: string
+  dataDpValidateIf?: string
+  dataDpValidateShouldEqual?: string
+  disabled?: boolean | null
+  hasIcon?: boolean
+  id: string
+  label?: {
+    bold: boolean
+    hide: boolean
+    hint: string
+    text: string
+    tooltip: string
+  };
+  maxlength?: boolean | string | null
+  minlength?: boolean | string | null
+  name?: string
+  pattern?: string
+  placeholder?: string
+  preventDefaultOnEnter?: boolean
+  readonly?: boolean | null
+  required?: boolean | null
+  size?: number | null
+  type?: string
+  value?: string
+  width?: string
+}
 
-  /**
-   * Full width by default; set to 'auto' to have no width defined.
-   * @deprecated Apply width to the parent element of DpInput.
-   */
-  width: {
-    type: String,
-    default: 'w-full'
-  }
-})
+// Use the interface with defineProps
+const props = defineProps<Props>();
 
 const emit = defineEmits(['blur', 'enter', 'focus', 'input'])
 
-const { value, maxlength, minlength, width, size, label } = toRefs(props)
+const {
+  ariaLabelledby = null,
+  autocomplete = '',
+  dataCounter = '',
+  dataCy = '',
+  dataDpValidateError = '',
+  dataDpValidateErrorFieldname = '',
+  dataDpValidateIf = '',
+  dataDpValidateShouldEqual = '',
+  disabled = null,
+  hasIcon = false,
+  id,
+  name = '',
+  pattern = '',
+  placeholder = '',
+  preventDefaultOnEnter = true,
+  readonly = null,
+  required = null,
+  type = 'text',
+} = props
+
+const {
+  label = {
+    bold: true,
+    hide: false,
+    hint: '',
+    text: '',
+    tooltip: ''
+  },
+  maxlength = null,
+  minlength = null,
+  size = null,
+  value = '',
+  width = 'w-full'
+} = toRefs(props)
```

Comment:
what is achieved with the toRefs here? i thought props are reactive by default?

Reply:
actually I didn't do it. It was like that before. I don't know why its like that. Git doesn't want to tell me why it was done like that.